### PR TITLE
feat: implement st1-phase-2 session prompts

### PR DESCRIPTION
**Pull request by arrnd**
Phase 2 Session Prompts Feature Development

Session 2.1 - query_location_proofs Tool Implementation:

Add comprehensive location proof querying with filtering capabilities
Implement REST API integration with /api/v0/location-proofs endpoint
Add parameter validation for chain, prover, limit, and offset filters
Include proper async handling and structured error responses
Support pagination and geospatial data processing

Session 2.2 - get_location_proof_by_uid Tool Implementation:

Implement specific attestation retrieval by unique identifier
Add UID format validation (66-character hex with 0x prefix)
Create robust error handling for not-found vs malformed cases
Include complete proof data structure with verification details
Add comprehensive test coverage for positive and negative cases

Session 2.3 - get_astral_config Tool & Error Handling Standardization:

Add Astral API configuration endpoint integration
Implement standardized error response format across all tools
Create consistent success/error structure with detailed context
Add enhanced logging for debugging and monitoring
Include chain and schema information retrieval capabilities

### CODE REVIEW

File: tests/test_server.py [hunk 1]
```diff
@@ -28,3 +28,51 @@ async def test_server_info_tool():
...
+    
+    # Test invalid UID format
+    result = await get_location_proof_by_uid("invalid_uid")
+    assert result["success"] is False
+    assert result["error"] == "validation_error"
+    
+    # Test short UID
+    result = await get_location_proof_by_uid("0x123")
+    assert result["success"] is False
+    assert result["error"] == "validation_error"
```
Comment (SethDocherty) [inline tests/test_server.py]: I don't see any tests here for valid UIDs? We should also test for valid cases to ensure that any modifications to the `get_location_proof_by_uid` method do not impact valid results.

Comment (SethDocherty) [reply tests/test_server.py]: Added the additional tests

File: astral_mcp_server/server.py [hunk 2]
```diff
@@ -117,10 +122,307 @@ async def get_server_info() -> Dict[str, Any]:
...
+    except httpx.HTTPStatusError as e:
+        error_msg = f"Configuration request failed with status {e.response.status_code}"
+        logger.error(f"{error_msg}: {e.response.text}")
+        return {
+            "success": False,
+            "error": "api_error",
+            "message": error_msg,
+            "details": {
+                "status_code": e.response.status_code,
+                "response_text": e.response.text[:500]
```
[the remark this answers is not in the document]
Comment (SethDocherty) [reply astral_mcp_server/server.py]: Took the suggestion into consideration and created a named constant

File: astral_mcp_server/server.py [hunk 3]
```diff
@@ -117,10 +122,307 @@ async def get_server_info() -> Dict[str, Any]:
...
+        error_msg = f"API request failed with status {e.response.status_code}"
+        logger.error(f"{error_msg}: {e.response.text}")
+        return {
+            "success": False,
+            "error": "api_error", 
+            "message": error_msg,
+            "details": {
+                "attempted_uid": uid,
+                "status_code": e.response.status_code,
+                "response_text": e.response.text[:500]
```
[the remark this answers is not in the document]
Comment (SethDocherty) [reply astral_mcp_server/server.py]: Took the suggestion into consideration and created a named constant

File: astral_mcp_server/server.py [hunk 4]
```diff
@@ -117,10 +122,307 @@ async def get_server_info() -> Dict[str, Any]:
...
+    except httpx.HTTPStatusError as e:
+        error_msg = f"API request failed with status {e.response.status_code}"
+        logger.error(f"{error_msg}: {e.response.text}")
+        return {
+            "success": False,
+            "error": "api_error",
+            "message": error_msg,
+            "details": {
+                "status_code": e.response.status_code,
+                "response_text": e.response.text[:500]  # Limit error text length
```
[the remark this answers is not in the document]
Comment (SethDocherty) [reply astral_mcp_server/server.py]: Took the suggestion into consideration and created a named constant

File: README.md
Comment (SethDocherty) [inline README.md:1]: The new MCP agent tools that were created were missing references in the main readme

File: tests/test_server.py
Comment (SethDocherty) [inline tests/test_server.py:1]: Test cases here are lacking and very basic.  It's good practice to consider the potential pitfalls and implications of each upstream/downstream component.

File: astral_mcp_server/server.py
Comment (SethDocherty) [inline astral_mcp_server/server.py:1]: After testing the experience using VSCode GitHub Copilot, I added additional functionality to extract geoJSON context from attestations, optimizing the Agents' response.

**Review by SethDocherty — APPROVED** [review]
Thanks for working on this @arrnd. You laid the bricks for a solid foundation, and I was able to get the basic functionality up and running. It wasn't immediately clear what new agent tools were available for me to use or how to utilize them. I did encounter a couple of cases where I wasn't getting any results back or an explanation of why. I was able to better identify the issues after updating some of the test cases.  

While you met the low-hanging MVP, there was ample opportunity for you to go the extra mile and make some solid additions, such as exploring MCP use cases,  the response UI, and starting to build some of the documentation.

After reviewing and expanding on the foundation you laid, I'm gonna go ahead and approve and merge the work
